main.tf

```diff
@@ -1,5 +1,5 @@
 resource "aws_vpn_connection" "default" {
-  count = "${var.create_vpn_connection ? 1 : 0}"
+  count = "${var.create_vpn_connection && length(var.tunnel1_inside_cidr) == 0 && length(var.tunnel2_inside_cidr) == 0 && length(var.tunnel1_preshared_key) == 0 && length(var.tunnel2_preshared_key) == 0 ? 1 : 0}"
```

Comment:
I would suggest to refactor the additional logic to a local variable with a name that expresses the intent.
Something like:
```
locals {
tunnel_details_not_specified = "${length(var.tunnel1_inside_cidr) == 0 && length(var.tunnel2_inside_cidr) == 0 && length(var.tunnel1_preshared_key) == 0 && length(var.tunnel2_preshared_key) == 0}"
}

...

resource "aws_vpn_connection" "default" {
  count = "${var.create_vpn_connection && local.tunnel_details_not_specified ? 1 : 0}"
  ...
}
```